packages/jazz-tools/src/tools/subscribe/CoValueCoreSubscription.ts

```diff
@@ -1,76 +1,196 @@
-import { CoValueCore, LocalNode, RawCoMap, RawCoValue } from "cojson";
+import { CoValueCore, LocalNode, RawCoID, RawCoMap, RawCoValue } from "cojson";
 
+/**
+ * Manages subscriptions to CoValue cores, handling both direct subscriptions
+ * and branch-based subscriptions with automatic loading and error handling.
+ *
+ * It tries to resolve the value immediately if already available in memory.
+ */
 export class CoValueCoreSubscription {
-  _unsubscribe: () => void = () => {};
-  unsubscribed = false;
-
-  value: RawCoMap | undefined;
+  private _unsubscribe: () => void = () => {};
+  private unsubscribed = false;
 
   constructor(
     public node: LocalNode,
     public id: string,
     public listener: (value: RawCoValue | "unavailable") => void,
     public skipRetry?: boolean,
+    public branch?: { name: string; ownerId?: string },
   ) {
-    const entry = this.node.getCoValue(this.id as any);
+    this.initializeSubscription();
+  }
+
+  /**
+   * Main entry point for subscription initialization.
+   * Determines the subscription strategy based on current availability and branch requirements.
+   */
+  private initializeSubscription(): void {
+    const source = this.node.getCoValue(this.id as any);
+
+    // If the CoValue is already available, handle it immediately
+    if (source.isAvailable()) {
+      this.handleAvailableSource(source);
+      return;
+    }
+
+    // If a specific branch is requested while the source is not available, attempt to checkout that branch
+    if (this.branch) {
+      this.handleBranchCheckout();
+    }
 
-    if (entry?.isAvailable()) {
-      this.subscribe(entry.getCurrentContent());
+    // If we don't have a branch requested, load the CoValue
+    this.loadCoValue();
+  }
+
+  /**
+   * Handles the case where the CoValue source is immediately available.
+   * Either subscribes directly or attempts to get the requested branch.
+   */
+  private handleAvailableSource(source: CoValueCore): void {
+    if (this.branch) {
```

Comment:
I would suggest inverting the condition order to simplify the readability, with the smaller implementation first:
```ts
if(!this.branch) {
  // No branch requested, subscribe directly to the source
  this.subscribe(source.getCurrentContent());
  return;
}

// ...
```